Web Inspector: Timelines: show a separate overview record bar row for each `WI.Target`

dcrousso opened the pull request:
#### e4b6167f9482ba564157c37005f095027deb3a3c
<pre>
Web Inspector: Timelines: show a separate overview record bar row for each `WI.Target`
<a href="https://bugs.webkit.org/show_bug.cgi?id=288687">https://bugs.webkit.org/show_bug.cgi?id=288687</a>

Reviewed by BJ Burg.

This will allow developers to more visually distinguish activity based on the `WI.Target` (e.g. is this page activity or in a `Worker`?).

* Source/WebInspectorUI/UserInterface/Views/ScriptTimelineOverviewGraph.js:
(WI.ScriptTimelineOverviewGraph):
(WI.ScriptTimelineOverviewGraph.prototype.reset):
(WI.ScriptTimelineOverviewGraph.prototype.layout):
(WI.ScriptTimelineOverviewGraph.prototype.layout.createBar):
(WI.ScriptTimelineOverviewGraph.prototype.updateSelectedRecord):
(WI.ScriptTimelineOverviewGraph.prototype._scriptTimelineRecordAdded):

* Source/WebInspectorUI/UserInterface/Views/ScriptTimelineOverviewGraph.css:
(.timeline-overview-graph.script &gt; .graph-row): Added.
(.timeline-overview-graph.script &gt; .graph-row:first-child:nth-last-child(1)): Added.
(.timeline-overview-graph.script &gt; .graph-row:first-child:nth-last-child(2), .timeline-overview-graph.script &gt; .graph-row:first-child:nth-last-child(2) ~ .graph-row): Added.
(.timeline-overview-graph.script &gt; .graph-row:first-child:nth-last-child(3), .timeline-overview-graph.script &gt; .graph-row:first-child:nth-last-child(3) ~ .graph-row): Added.
(.timeline-overview-graph.script &gt; .graph-row:first-child:nth-last-child(4), .timeline-overview-graph.script &gt; .graph-row:first-child:nth-last-child(4) ~ .graph-row): Added.
(.timeline-overview-graph.script &gt; .graph-row &gt; .timeline-record-bar): Renamed from `.timeline-overview-graph.script &gt; .timeline-record-bar`.
(.timeline-overview-graph.script &gt; .graph-row &gt; .timeline-record-bar &gt; .segment): Renamed from `.timeline-overview-graph.script &gt; .timeline-record-bar &gt; .segment`.
Adjust the height of the record bar row based on how many of them there are.

* Source/WebInspectorUI/UserInterface/Views/TimelineOverview.css:
(.timeline-overview &gt; .graphs-container &gt; .timeline-overview-graph):
(.timeline-overview:not(.frames) &gt; .graphs-container &gt; .timeline-overview-graph:not(:first-child)):
Use CSS variables to allow future developers to search for uses of it to know what to also modify when changing the value.

Canonical link: <a href="https://commits.webkit.org/292169@main">https://commits.webkit.org/292169@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ab08d7ad904cdaf3b0619bfe85d0932699c2f2bf

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/92565 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/12110 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/1721 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/97550 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/43072 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/12389 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/20568 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70899 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/28346 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/95567 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/9393 "Found 1 new test failure: fast/forms/ios/focus-input-in-fixed.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/83808 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/51231 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/9089 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/1447 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/42403 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/79405 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/1387 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/99576 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/19616 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/14463 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79903 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/19866 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/79691 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/79185 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20247 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/23714 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/1018 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/12628 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/19600 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/24772 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/19287 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/22747 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/21027 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

burg reviewed (changes requested):
The CSS looks neat. My primary concern is that this may have unintended consequences for inspecting sites that have more than a few workers- do the rows become unusably short? Is it still possible to click on the pills to select rows?

Once we have a target per frame, this will certainly require more coalescing to mitigate the effects of having too many targets.

However, I'd like to move forward in the direction of this change, least because it adds target support to ScriptTimelineOverviewGraph. Could you put the rendering of >1 bar behind an Experimental checkbox for now, so we can continue to iterate?

burg reviewed (approved):
Did you test that there is no regression when the experimental setting is toggled off?

In general I'd like to include a few sentences about how to go about testing each change that doesn't have a test case. This allows us to identify testing gaps and supports non-Web Inspector QA people in understanding how to test a change.

dcrousso committed:
Web Inspector: Timelines: show a separate overview record bar row for each `WI.Target`
https://bugs.webkit.org/show_bug.cgi?id=288687

Reviewed by BJ Burg.

This will allow developers to more visually distinguish activity based on the `WI.Target` (e.g. is this page activity or in a `Worker`?).

* Source/WebInspectorUI/UserInterface/Views/ScriptTimelineOverviewGraph.js:
(WI.ScriptTimelineOverviewGraph):
(WI.ScriptTimelineOverviewGraph.prototype.reset):
(WI.ScriptTimelineOverviewGraph.prototype.layout):
(WI.ScriptTimelineOverviewGraph.prototype.layout.createBar):
(WI.ScriptTimelineOverviewGraph.prototype.updateSelectedRecord):
(WI.ScriptTimelineOverviewGraph.prototype._scriptTimelineRecordAdded):

* Source/WebInspectorUI/UserInterface/Views/ScriptTimelineOverviewGraph.css:
(.timeline-overview-graph.script > .graph-row): Added.
(.timeline-overview-graph.script > .graph-row:first-child:nth-last-child(1)): Added.
(.timeline-overview-graph.script > .graph-row:first-child:nth-last-child(2), .timeline-overview-graph.script > .graph-row:first-child:nth-last-child(2) ~ .graph-row): Added.
(.timeline-overview-graph.script > .graph-row:first-child:nth-last-child(3), .timeline-overview-graph.script > .graph-row:first-child:nth-last-child(3) ~ .graph-row): Added.
(.timeline-overview-graph.script > .graph-row:first-child:nth-last-child(4), .timeline-overview-graph.script > .graph-row:first-child:nth-last-child(4) ~ .graph-row): Added.
(.timeline-overview-graph.script > .graph-row > .timeline-record-bar): Renamed from `.timeline-overview-graph.script > .timeline-record-bar`.
(.timeline-overview-graph.script > .graph-row > .timeline-record-bar > .segment): Renamed from `.timeline-overview-graph.script > .timeline-record-bar > .segment`.
Adjust the height of the record bar row based on how many of them there are.

* Source/WebInspectorUI/UserInterface/Views/TimelineOverview.css:
(.timeline-overview > .graphs-container > .timeline-overview-graph):
(.timeline-overview:not(.frames) > .graphs-container > .timeline-overview-graph:not(:first-child)):
Use CSS variables to allow future developers to search for uses of it to know what to also modify when changing the value.

Canonical link: https://commits.webkit.org/292169@main
